secrets/package.go

```
@@ -2,54 +2,34 @@ package secrets
 
 import (
 	"context"
-	"flag"
 	"os"
-	"strings"
-
-	"github.com/aws/aws-sdk-go-v2/config"
-	"github.com/go-errors/errors"
 )
 
-// DefaultAWSSecretsManager is a public *AWSSecretsManager used for package level methods.
-var DefaultAWSSecretsManager = getInstance()
-
-func getInstance() *AWSSecretsManager {
-	var client AWSSecretsManagerClient
-
-	if isTestMode() {
-		client = newTestRunnerClient()
-	} else {
-		region := os.Getenv("AWS_REGION")
-		if region == "" {
-			err := errors.Errorf("missing AWS_REGION environment variable")
-			panic(err)
-		}
-
-		cfg, err := config.LoadDefaultConfig(context.Background(), config.WithRegion(region))
-		if err != nil {
-			err := errors.Errorf("error loading default aws sdk config, err='%w'\n", err)
-			panic(err)
-		}
-
-		client = newSecretManagerClient(cfg)
-	}
-	return NewAWSSecretsManagerWithClient(client)
+// Secrets can be mocked by clients for testing purposes.
+type Secrets interface {
+	Get(ctx context.Context, secretKey string) (string, error)
 }
 
+// DefaultAWSSecretsManager is a public *AWSSecretsManager used for package level methods.
+var DefaultAWSSecretsManager Secrets = nil
+
 // Get retrives the secret from AWS SecretsManager.
 func Get(ctx context.Context, secretKey string) (string, error) {
```

Since these methods already return an error, I reckon it is better to return the error here than panicking.

---

Agree - I've fixed this in the latest commit.